benchmarks/benchmarks/sparse_linalg_lobpcg.py

```diff
@@ -66,45 +22,119 @@ def __init__(self):
         self.time_mikota.__func__.setup = self.setup_mikota
 
         self.time_sakurai.__func__.params = list(self.params)
-        self.time_sakurai.__func__.params[0] = [50, 400]
+        self.time_sakurai.__func__.params[0] = [50, 100]
         self.time_sakurai.__func__.setup = self.setup_sakurai
 
+        self.time_sakuraii.__func__.params = list(self.params)
+        self.time_sakuraii.__func__.params[0] = [500, 1000]
+        self.time_sakuraii.__func__.setup = self.setup_sakuraii
+
+
     def setup_mikota(self, n, solver):
         self.shape = (n, n)
-        self.A, self.B = _mikota_pair(n)
-
-        if solver == 'eigh' and n >= 512:
+        mik = Mikota_pair(n)
+        mik_k = mik.k
+        mik_m = mik.m
+        self.Ac = mik_k
+        self.Aa = mik_k.toarray()
+        self.Bc = mik_m
+        self.Ba = mik_m.toarray()
+        self.Ab = mik_k.tobanded()
+        self.eigenvalues = mik.eigenvalues
+
+        if solver == 'eigh' and n > 512:
             # skip: slow, and not useful to benchmark
             raise NotImplementedError()
 
     def setup_sakurai(self, n, solver):
         self.shape = (n, n)
-        self.A, self.B, all_eigenvalues = _sakurai(n)
-        self.A_dense = self.A.A
-        self.B_dense = self.B.A
+        sakurai_obj = Sakurai(n, dtype='int')
+        self.A = sakurai_obj
+        self.Aa = sakurai_obj.toarray()
+        self.eigenvalues = sakurai_obj.eigenvalues
 
+    def setup_sakuraii(self, n, solver):
+        self.shape = (n, n)
+        sakurai_obj = Sakurai(n)
+        self.A = sakurai_obj.tobanded()
+        self.eigenvalues = sakurai_obj.eigenvalues
+    
     def time_mikota(self, n, solver):
+        def a(x):
+            return cho_solve_banded((c, False), x)
         m = 10
+        ee = self.eigenvalues[:m]
+        tol = m * n * n * n* np.finfo(float).eps
+        rng = np.random.default_rng(0)
+        X = rng.normal(size=(n, m))
         if solver == 'lobpcg':
-            X = np.random.rand(n, m)
-            X = orth(X)
-            LorU, lower = cho_factor(self.A, lower=0, overwrite_a=0)
-            M = LinearOperator(self.shape,
-                               matvec=partial(_precond, LorU, lower),
-                               matmat=partial(_precond, LorU, lower))
-            eigs, vecs = lobpcg(self.A, X, self.B, M, tol=1e-4, maxiter=40)
+            c = cholesky_banded(self.Ab.astype(np.float32))
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                el, _ = lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
+                               maxiter=40, largest=False)
+            accuracy = max(abs(ee - el) / ee)
+            assert accuracy < tol
```

Comment:
At least at one point in all those benchmarks (preferably at least once per `test_*`), we should note the following
```suggestion
            # ensure that we are benchmarking a consistent outcome;
            # (e.g. if the code wasn't able to find a solution at all,
            # the timing of the benchmark would become useless).
            assert accuracy < tol
```

Reply:
Actually, it might be good to define a string that gets used in all the asserts as a message in case of failure:
```python
msg = ("the benchmark code did not converge as expected, "
       "the timing is therefore useless")
assert accuracy < tol, msg